### PR TITLE
Add Configuring Docker article

### LINKED ISSUE #12088: Docker Docs : Add information about configuring the docker daemon after installation
All the [Installation docs](https://docs.docker.com/installation/) cover how to install docker in detail, but none of them talk about configuring the docker daemon after the installation.

There should be a short section in there describing the config process. For example, in Ubuntu, the daemon can be configured by adding options in `/etc/default/docker` and in CentOS/RHEL, the daemon can be configured by editing `/etc/sysconfig/docker`

There is [one article](https://docs.docker.com/articles/systemd/) which covers this, but it is very specific. It would be nice to have a short config article after every installation step for various Operating Systems

### PR DESCRIPTION
**Pull request by ankushagarwal**
Fixes #12088

### CODE REVIEW

**Comment by ankushagarwal** [comment]
@moxiegirl @thaJeztah 
Can you review this? I have only added Ubuntu for now, will add others after feedback.

**Comment by vdemeester** [comment]
@ankushagarwal There should be something about docker configuration with systemd for example (where you update the ``.service`` file, and no ``/etc/default/docker``)

**Comment by ankushagarwal** [comment]
I will. I just want to get Ubuntu reviewed first, which uses upstart.

**Comment by vdemeester** [comment]
ah ok ;-)

File: docs/sources/articles/configuring.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+page_title: Configuring Docker
+page_description: Configuring the Docker daemon on various distributions
+page_keywords: docker, daemon, configuration
+
+# Configuring Docker on various distributions
```
Comment (moxiegirl) [inline docs/sources/articles/configuring.md]: Thanks for the contribution.  Very nice.   I think your configuring section will be easier to follow as a procedure.

---> https://gist.github.com/moxiegirl/52bc6e3c921d1f4bbb12

page_title: Configuring Docker
page_description: Configuring the Docker daemon on various distributions
page_keywords: docker, daemon, configuration

# Configuring Docker on various distributions

After successfully installing Docker, the `docker` daemon runs with it's default
configuration. You can configure the daemon by passing configuration
flags to it directly when you start it.  

In a production environment, system administrators typically configure the
daemon to start and stop according to an organization's requirements.  In this
case, the administrator configures a process manager such as SysVinit, Upstart,
or `systemd` to manage the `docker` daemon's start and stop. Some of the daemons
options are:

| Flag                  | Description                                               |
|-----------------------|-----------------------------------------------------------|
| `-D`, `--debug=false` | Enable or disable debug mode.  By default, this is false. |
| `-H`,`--host=[]`      | Daemon socket(s) to connect to.                           |
| `--tls=false`         | Enable or disable TLS. By default, this is false.         |

The command line reference has the [complete list of daemon flags](/reference/commandline/cli/#daemon).

## Ubuntu

After successfully [installing Docker for Ubuntu](/installation/ubuntulinux/), you can check the
running status using Upstart in this way:

    $ sudo status docker
    docker start/running, process 989

You can start/stop/restart `docker` using

    $ sudo start docker

    $ sudo stop docker

    $ sudo restart docker


### Configuring Docker

You configure the `docker` daemon in the `/etc/default/docker` file on your
system.  You do this by specifying values in a `DOCKER_OPTS` variable. 
To configure Docker options:

1. Log into your system as a user with `sudo` or `root` privileges.

2. If you don't have one, create the `/etc/default/docker` file in your system. 

	Depending on how you installed Docker, you may already have this file.

3. Open the file with your favorite editor.

		$ sudo vi /etc/default/docker
		
4. Add a `DOCKER_OPTS` variable with the following options.

	``` 
	 DOCKER_OPTS=" --dns 8.8.8.8 -D --tls=false -H tcp://0.0.0.0:2375 "
	```
	
	These options : 

	- Set `dns` server for all containers
	- Enable `-D` (debug) mode 
  - Set `tls` to false
  - Listen for connections on `tcp://0.0.0.0:2375`
  
5. Save and close the file.

6. Restart the `docker` daemon.

 		 $ sudo restart docker

7. Verify that the daemon is running as specified wit the `ps` command.

		$ ps aux | grep docker | grep -v grep

**Comment by moxiegirl** [comment]
@ankushagarwal One thing I did notice is that 1.6 on my version of ubuntu created the configuration file but it looks different.

```
# Docker Upstart and SysVinit configuration file

# Customize location of Docker binary (especially for development testing).
#DOCKER="/usr/local/bin/docker"

# Use DOCKER_OPTS to modify the daemon startup options.
#DOCKER_OPTS="--dns 8.8.8.8 --dns 8.8.4.4"

# If you need Docker to use an HTTP proxy, it can also be specified here.
#export http_proxy="http://127.0.0.1:3128/"

# This is also a handy place to tweak where Docker's temporary files go.
#export TMPDIR="/mnt/bigdrive/docker-tmp"
```

Your procedure doesn't touch on removing the second `dns` specification. Do recommend removing that? If so, should probably mention it.

**Comment by ankushagarwal** [comment]
The `#DOCKER_OPTS="--dns 8.8.8.8 --dns 8.8.4.4"` line is just a place holder. It is neither recommended or unrecommended to use particular DNS servers. 8.8.8.8 and 8.8.4.4 are just two popular DNS servers and it is unlikely that people will actually use these options for the production use case where they have their own internal DNS servers.

But for clarity, I will add 8.8.4.4 to my doc as well.

File: docs/sources/articles/configuring.md (hunk 1 above)
Comment (ankushagarwal) [inline docs/sources/articles/configuring.md]: Thanks, this is easier to follow. Will update.

**Comment by ankushagarwal** [comment]
Updated my patch and here is the diff from your patch : https://gist.github.com/ankushagarwal/aff84863609515dadafa/revisions

**Comment by ankushagarwal** [comment]
Will add other systems soon

**Comment by moxiegirl** [comment]
Thanks @ankushagarwal much appreciate the contribution!  LGTM

ping @docker/docs-owners

File: docs/sources/articles/configuring.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	``` 
+	 DOCKER_OPTS=" --dns 8.8.8.8 --dns 8.8.4.4 -D --tls=false -H tcp://0.0.0.0:2375 "
+	```
+	
+These options : 
+
+- Set `dns` server for all containers
+- Enable `-D` (debug) mode 
+- Set `tls` to false
+- Listen for connections on `tcp://0.0.0.0:2375`
```
Comment (SvenDowideit) [inline docs/sources/articles/configuring.md]: @ankushagarwal in your future PR's could you please add notes to say something along the lines of never run your Docker daemon using tls=false - and point the reader at the https Docker daemon document?

using the unencrypted socket means that server is a phishing target that exposes root access to the box - so 'very bad mkay' 

but LGTM, given you're working on further updates.

**Comment by SvenDowideit** [comment]
LGTM, with more to come in later updates. (including links from the installation docs to their relevant config sections

awesome - thank you!

**Comment by fredlf** [comment]
This is great. Thanks for doing it. My only comment (too late) is that we should think about disambiguating "Configure Docker" to something like "Configure Docker Engine".

**Comment by ankushagarwal** [comment]
@SvenDowideit : Makes sense. Will replace that option with true
@fredlf : Good suggestion, will update in the next PR

**Pull request merged**